subiquity/tests/api/test_api.py

```diff
@@ -110,7 +110,7 @@ async def request(
         async with self.session.request(
             method, f"http://a{query}", data=data, params=params, headers=headers
         ) as resp:
-            print(unquote(str(resp.url)))
+            # print(unquote(str(resp.url)))
```

Comment:
I disagree with this change and the one below, I generally want to see the response information when these tests fail. Is there some other instance in which they're cluttering test output? Unless they fail I only get them when passing `-s` to pytest.

Reply:
We can drop this commit if preferred. My main concern is that it is generating the output intermingled with the regular Python unittest output. I think it would be better to have it use some appropriate logging function. In any case, this commit isn't really related to the overall PR. I made it originally when I thought test_api.py might be the appropriate place to put the new unit test that I did add. If you want it dropped, I'll rebase with this commit removed.

Reply:
I should add, all the output I silences was happening even in the successful cases when no verbosity was turned on. That is the main reason I felt to comment it out. Any noise I see in what should be a successful run usually implies that something needs to be fixed to me.

Reply:
Yes let's revisit this separately I think, thanks. It's pretty convenient behavior with pytest, which is our primary test runner.

Reply:
Interesting, I wonder if `pytest` handles stdout differently. I was using `unittest` as the test runner and did not realize that `pytest` supported running `unittest`-style tests. It was not clear how these tests are normally run as they don't seem to be run from the "tox.ini" configuration. I had to explicitly reference the full path to the class as I seemed to have an incomplete environment to run the full test suite. However, once I discovered the scripts test-runner, that ended up being a more straight-forward way to write the test I wanted. I'll drop this for now.

Reply:
OK, now that I better understand how the test framework works, I ran it with `make api` and saw it cleanly execute all 82 tests. If one fails, then it dumps all the captured output from that failed test in the end-of-test report. Now, if I replace `-m pytest -n auto` with `-m unittest`, it still successfully reports all 82 tests were successful, but completely blasts the terminal with lots of noisy output. When the tests are run correctly, there's no issue. Maybe it's time I switch testing frameworks for my projects...